### ISSUE TITLE
[package] qt/5.15.2: linkage errors on Network module on Windows and Mac

### ISSUE THREAD

**Issue by AndreyMlashkin** [conan-io/conan-center-index#6071]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **qt/5.15.2**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 19**
  * Conan version: **conan 1.37.2**
  * Python version: **Python 3.7.4**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.runtime=MT
compiler.version=16
build_type=Release
[options]
*:fPIC=True
*:shared=False
[build_requires]
[env]
CONAN_USE_ALWAYS_SHORT_PATHS=True
```

### Steps to reproduce (Include if Applicable)
install any package, using Qt5 Network module

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
     Creating library C:/.conan/7147bc/1/lib/XXX.lib and object C:/.conan/7147bc/1/lib/XXX.exp
Qt5Network.lib(qnetconmonitor_win.obj) : error LNK2019: unresolved external symbol __imp_ConvertInterfaceIndexToLuid referenced in function "public: bool __cdecl QNetworkConnectionEvents::setTarget(class QNetworkInterface const &)" (?setTarget@QNetworkConnectionEvents@@QEAA_NAEBVQNetworkInterface@@@Z) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2001: unresolved external symbol __imp_ConvertInterfaceIndexToLuid [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetconmonitor_win.obj) : error LNK2019: unresolved external symbol __imp_ConvertInterfaceLuidToGuid referenced in function "public: bool __cdecl QNetworkConnectionEvents::setTarget(class QNetworkInterface const &)" (?setTarget@QNetworkConnectionEvents@@QEAA_NAEBVQNetworkInterface@@@Z) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2019: unresolved external symbol __imp_GetNetworkParams referenced in function "public: static class QString __cdecl QHostInfo::localDomainName(void)" (?localDomainName@QHostInfo@@SA?AVQString@@XZ) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2019: unresolved external symbol __imp_GetAdaptersAddresses referenced in function "class QList<class QNetworkInterfacePrivate *> __cdecl interfaceListing(void)" (?interfaceListing@@YA?AV?$QList@PEAVQNetworkInterfacePrivate@@@@XZ) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2019: unresolved external symbol __imp_ConvertInterfaceNameToLuidW referenced in function "public: static unsigned int __cdecl QNetworkInterfaceManager::interfaceIndexFromName(class QString const &)" (?interfaceIndexFromName@QNetworkInterfaceManager@@SAIAEBVQString@@@Z) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2019: unresolved external symbol __imp_ConvertInterfaceLuidToNameW referenced in function "class QHostAddress __cdecl addressFromSockaddr(struct sockaddr *)" (?addressFromSockaddr@@YA?AVQHostAddress@@PEAUsockaddr@@@Z) [C:\.conan\7147bc\1\XXX.vcxproj]
Qt5Network.lib(qnetworkinterface_win.obj) : error LNK2019: unresolved external symbol __imp_ConvertInterfaceLuidToIndex referenced in function "public: static unsigned int __cdecl QNetworkInterfaceManager::interfaceIndexFromName(class QString const &)" (?interfaceIndexFromName@QNetworkInterfaceManager@@SAIAEBVQString@@@Z) [C:\.conan\7147bc\1\XXX.vcxproj]
LIBCMT.lib(exe_winmain.obj) : error LNK2019: unresolved external symbol WinMain referenced in function "int __cdecl __scrt_common_main_seh(void)" (?__scrt_common_main_seh@@YAHXZ) [C:\.conan\7147bc\1\XXX.vcxproj]
C:\.conan\7147bc\1\bin\XXX.exe : fatal error LNK1120: 8 unresolved externals [C:\.conan\7147bc\1\XXX.vcxproj]
XXX/v14.27.6-1-gf87cf5703@gitlab/testing: 
libgettext/0.20.1: WARN: Recipe 'libgettext/0.20.1' is deprecated in favor of 'gettext'. Please, consider changing your requirements.
WARN: pictofitcore/v28.7.5@gitlab/release: requirement xremesh/1.0.6 overridden by XXX/v14.27.6-1-gf87cf5703@gitlab/testing to xremesh/1.0.7 
WARN: ceres-solver/2.0.0: requirement eigen/3.3.8 overridden by pictofitcore/v28.7.5@gitlab/release to eigen/3.3.9 
XXX/v14.27.6-1-gf87cf5703@gitlab/testing: ERROR: Package '53f956726bf49085398d845274ea029eae8a024b' build failed
XXX/v14.27.6-1-gf87cf5703@gitlab/testing: WARN: Build folder C:\.conan\7147bc\1
ERROR: XXX/v14.27.6-1-gf87cf5703@gitlab/testing: Error in build() method, line 147
	cmake.build()
	ConanException: Error 1 while executing cmake --build C:\.conan\7147bc\1 --config Release -- /m:2 /verbosity:minimal
[XXX/v14.27.6-1-gf87cf5703@gitlab/testing] recipe could not be compiled
```

</details>

**Comment by AndreyMlashkin** [comment]
Attached PR demonstrates the problem

**Comment by ericLemanissier** [comment]
this was solved with #6072
